### PR TITLE
More documentation and traits for `Locale`

### PR DESCRIPTION
**Pull request by pitdicker**
For chrono we re-export the `Locale` type from `pure-rust-locales`, and are going to include in in the documentation on docs.rs with the next release (still behind the `unstable-locales` feature however).

It would be nice if the enum would have more documentation for the variants and itself.

I also added the `Default`, `Eq` and `Hash` traits, and made `Display` en `Debug` implementations match the `TryFrom<&str>` implementation.

### CODE REVIEW

File: generate-api/src/generator.rs
Comment (pitdicker) [inline generate-api/src/generator.rs:502]: This license note seems quite out of place :smile:.
When a someone browses the documentation of an MIT-licensed library and then seeds "glibc" and "GNU" I expect this will be a question.

**Comment by pitdicker** [comment]
afbeelding

**Review by cecton — COMMENTED** [review]
Awesome thanks!! Sorry for the delay I completely forgot to answer in the first place

I just have that small question for Default but other than that it's great

**Comment by cecton** [comment]
If you want I invite you to co-own this crate? Right now there is only me on this and I'm not too comfortable with the bus factor. (We could also move the project to chrono I really don't mind.)

File: generate-api/src/generator.rs
Comment (pitdicker) [reply generate-api/src/generator.rs:504]: Good point, that is tricky.

According to https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap07.html#tag_07_02:

> For C-language programs, the POSIX locale shall be the default locale when the [setlocale()](https://pubs.opengroup.org/onlinepubs/009695399/functions/setlocale.html) function is not called.

In chrono I added a `default_locale()` function: https://github.com/chronotope/chrono/blob/0.4.x/src/format/locales.rs#L5 to return `POSIX`.

File: generate-api/src/generator.rs
Comment (pitdicker) [reply generate-api/src/generator.rs:502]: Good to see the complete context, https://github.com/chronotope/chrono/pull/453#issue-655205860 had the important bit of it :+1:.

**Comment by pitdicker** [comment]
> If you want I invite you to co-own this crate? Right now there is only me on this and I'm not too comfortable with the bus factor. (We could also move the project to chrono I really don't mind.)

Thank you! Let me think about this a bit. cc @djc.

**Comment by djc** [comment]
Given that development here already seems to be driven by chrono's needs, I think it could make sense to move this into the chronotope org.

File: generate-api/src/generator.rs
Comment (cecton) [reply generate-api/src/generator.rs:504]: I would be happy with posix

**Comment by cecton** [comment]
There is a high probability that this project will be used on other crates though. In this comment (https://github.com/cecton/pure-rust-locales/pull/7#issuecomment-1715325212) you can see that the author intends to use it for number formatting which is part of i18n efforts.

But I think its main use at the moment is chrono and it will be great to have it there for maintenance reasons. In the end it's very small and very low maintenance, there is no need for me to be involved but I'm happy to help whenever needed. So you could add me as maintainer on the project itself but not on the organization.

Ideally I believe it is important to make people realize somehow (through documentation maybe?) that this isn't necessary. All of this is available through the libc for which the implementation varies from one platform to the other (and from one libc to the other because musl seems to have no i18n thingy whatsoever). So they also **not** use chrono for string formatting but strftime directly. The only difference here is that if you use pure-rust-locales, the result will be consistent across platform and they don't require setupping up locales on the machine (which can be useful on cloud). (Which is why I made the crate [libc-strftime](https://crates.io/crates/libc-strftime))

**Comment by djc** [comment]
Not entirely sure what this is blocking on? If you add me as an admin for this repo I think I can move it into the chronotope org.

**Comment by cecton** [comment]
I couldn't find the button :sweat_smile: I sent it to you instead

**Comment by djc** [comment]
Yeah, I meant adding me as an admin to your repo, so I could transfer it. But this works, too! We're live in the chronotope org now.

**Comment by pitdicker** [comment]
Sorry I so completely dropped out. It is still okay to merge this PR?

**Comment by djc** [comment]
This doesn't have any approvals yet, so I guess not?

**Comment by pitdicker** [comment]
Only:

> I just have that small question for Default but other than that it's great

Which we then discussed with POSIX as a good default.

**Comment by djc** [comment]
Would be good to get an explicit approval from @cecton.

**Comment by pitdicker** [comment]
Agreed :smile:.

File: generate-api/src/generator.rs
Comment (cecton) [inline generate-api/src/generator.rs:534]: ahh that's the default

**Review by cecton — APPROVED** [review]

**Comment by pitdicker** [comment]
Thank you! Lets try my first merge in this crate :smile:.

**Pull request merged**
